docs/enhancements/boot-status-reporter.md

```diff
@@ -0,0 +1,129 @@
+---
+title: boot-status-reporter
+authors:
+  - "@nmagnezi"
+creation-date: 2022-09-22
+last-updated: 2022-09-22
+---
+
+# Assisted Boot Reporter
+
+## Summary
+
+Create a new reporting service that runs as early as possible when hosts are booted from disk to collect additional 
+information and send it to `assisted-service`.
+
+## Motivation
+
+There are multiple occurrences of clusters that fail to install at the stage where
+hosts manage to pull ignition but fail to get to a point where the `assisted-controller`
+pod runs and calls home to `assisted-service` with logs.
+
+With that missing information, there's not a whole lot we can do to understand what
+happened with those installations.
+
+
+### Goals
+
+1. Gather information for a host boot stage to improve our ability to debug failures and better inform 
+   users about the host progress stages (first boot, second boot, etc).
+
+
+2. The new service should be thoroughly cleaned up from the hosts and should never fail installations by itself.
+   If the boot reporter service fails to run or report logs, the cluster installation should continue, 
+   and the new host progress stages should be skipped in such a scenario.
+
+## Proposal
+
+
+### Changes To Assisted Service
+For cases as mentioned in `Motivation`, we need a new service (start via `systemd`) soon as possible after hosts
+are booting from the disk and contact `assisted-service` to:
+
+1. Change the installation stage to a new `Booted with local ignition` Between `Configuring`
+   and  `Waiting for control plane`. This stage refers to a boot with pointer ignition.
+
+
+2. Change the installation stage to a new `Booted with control plane ignition` Between
+   `Waiting for control plane` and  `Joined`.
+
+
+3. For each of the above-mentioned, Read the current host stage from `assisted-service`, to determine if:
+   - the service runs on the first boot :arrow_right: collect logs and report.
+   - or the second boot :arrow_right: collect logs, report, and self-cleanup after a configurable threshold.
+
+### Information Needed And Changes To The Discovery Agent
+
+In order to communicate with `assisted-service`, The assisted boot reported service needs to know:
+1. The `assisted-service` IP address and port.
+2. The pull-secret to use for agent authentication.
+3. The InfraEnv ID and host ID
+
+It will need this information starting from the first boot, which is a pointer ignition-based boot.
+
+For that, we need to modify the discovery agent to store this information on disk in a pre-defined location for the assisted boot reporter service to use.
```

Comment:
Why not embed the files in the installed ignition?

Reply:
I think that was the intention here, probably requires rephrasing.

Reply:
Still testing this. However, I had chat with @carbonin about this, and he thinks we should avoid it.
Instead, he proposed to have a script file downloaded via assisted-service API (probably a bash script).
I'll complete the test anyhow but mentioned this in case Nick would want to expand on this.

Reply:
Yes, but my suggestion was just to avoid an incredibly large payload of base64 data in ignition.
Ignition itself also allows files to be created using regular http urls so I'm suggesting to download whatever script/executable thing we use with that format rather than embedding directly into the pointer ignition.

I also suggested bash because it's the most portable.
I don't think we want to get into the business of compiling go programs for every architecture we might soon support.

I think there's a lot of room for discussion on how to go about distributing whatever this service is, and a lot of it depends on how complicated it ends up being.